client-tags/display-name.md

```diff
@@ -0,0 +1,104 @@
+    period: "2021"
+    email: "james@irccloud.com"
+---
+
+## Notes for implementing work-in-progress version
+
+This is a work-in-progress specification.
+
+Software implementing this work-in-progress specification MUST NOT use the
+unprefixed `+display-name` tag name. Instead, implementations SHOULD use the
+`+draft/display-name` tag name to be interoperable with other software
+implementing a compatible work-in-progress version.
+
+The final version of the specification will use an unprefixed tag name.
+
+## Introduction
+
+This specification defines a client-only message tag to indicate a display name to use for the attached message, separate from the nickname or realname of the user.
+
+## Motivation
+
+This tag provides a way to provide further temporary context about the originator of a message. Example use cases include bots that relay messages from external contexts, or in role playing environments.
+
+## Architecture
+
+For the purpose of this specification, a message is defined as a `PRIVMSG`, `NOTICE`, `TAGMSG`, or [`batch`][batch] end.
+
+### Dependencies
+
+Clients wishing to use this tag MUST negotiate the [`message-tags`][tags] capability with the server.
+
+### Format
+
+Display names are indicated by a message with a `+draft/display-name` tag using the client-only prefix with the value being the desired name.
+
+## Client implementation considerations
+
+This section is non-normative.
+
+This client tag is not meant as a method of impersonating other IRC users. Take care to make it clear to users when a display name is being used. Consider formatting display names differently to nicknames and making the original nickname visible to the user.
+
+This client tag is also not meant to be used as a persistent display name for users. The [metadata framework][metadata] would be more suitable for such a use case.
+
+### Fallback string removal
+
+Clients might wish to send messages that begin with a usable fallback for other clients that don't understand the display name tag. A common existing method of indicating a display name is be to include the desired name at the start of a message, surrounded by angle brackets, or other forms of parenthesis.
+
+To support this elegantly, receiving clients that support the display name might wish to remove such a prefix if it matches the display name tag.
+
+For instance, the following message has a display name tag with the value `charles` and the message begins with the fallback string `<charles> `.
+
+```
+@+draft/display-name=charles :bot!bot@bot PRIVMSG #channel :<charles> Hello from outside IRC
+```
+
+A receiving client could detect this and remove the fallback string before displaying the message to the user. An example regular expression for detecting such a fallback might be:
+
+```
+^[<`]DISPLAYNAME[>`]\s
+```
+
+Where `DISPLAYNAME` is substituted for the display name of the message, appropriately escaped for use in a regular expression. This would match angle brackets or backticks
```

Comment:
So... No `>` or backtick followed by a space allowed in display names?

Reply:
Why not? as long as the stuff in there matches the displayname tag you can still match e.g. <my> cool> display` name>

Reply:
Oh yes indeed. I didn't think of matching them.